src/eosjs2-api.ts

```diff
@@ -93,26 +93,41 @@ export class Api {
         }));
     }
 
-    async pushTransaction({ blocksBehind, expireSeconds, actions, ...transaction }: any) {
-        let info: GetInfoResult;
-        if (!this.chainId) {
-            info = await this.rpc.get_info();
-            this.chainId = info.chain_id;
-        }
-        if (blocksBehind !== undefined && expireSeconds !== undefined) {
-            if (!info)
-                info = await this.rpc.get_info();
-            let refBlock = await this.rpc.get_block(info.head_block_num - blocksBehind);
-            transaction = { ...ser.transactionHeader(refBlock, expireSeconds), ...transaction };
-        }
-        transaction = { ...transaction, actions: await this.serializeActions(actions) };
-        let serializedTransaction = this.serializeTransaction(transaction);
-        let availableKeys = await this.signatureProvider.getAvailableKeys();
-        let requiredKeys = await this.authorityProvider.getRequiredKeys({ transaction, availableKeys });
-        let signatures = await this.signatureProvider.sign({ chainId: this.chainId, requiredKeys, serializedTransaction: serializedTransaction });
-        return await this.rpc.push_transaction({
-            signatures,
-            serializedTransaction,
-        });
+    async fillTaposFields({ blocksBehind, expireSeconds, actions, ...transaction }: any) {
```

Comment:
All new methods should have unit tests.

Reply:
As far as I know, there are currently no tests for this repository.  If that's the case, it might be better to create a separate PR for all of the unit tests before/after this one.  I'd be happy to implement the unit tests for this repo if I was given some direction/goals.